### PR TITLE
JSAPI geocode / reverse geocode

### LINKED ISSUE #1442: Geocoding et reverse geocoding
fonction de geocoding permettant de trouver les coordonnées d’une adresse (geocode)
fonction de “reverse geocoding” permettant de trouver l’adresse la plus proche d’une coordonnée (reversegeocode)

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #1442 
@tonio Please review.

### CODE REVIEW

File: jsapi/examples/geocode.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  zip: 1420,
+  locality: 'luxembourg'
+}, function(position) {
+  map.showMarker({
+    position: position,
+    positionSrs: 4326,
+    autoCenter: true,
+    positioning: 'center-center',
+    iconURL: 'http://apps.geoportail.lu/exemple_api/exemplesWikiGeoAPI/lion.png',
+    html: address
```
Comment (tonio) [inline jsapi/examples/geocode.js]: never defined?

File: jsapi/externs/luxx.js [hunk 2]
```diff
@@ -333,3 +333,23 @@ luxx.State.prototype.opacities;
  * @type {string}
  */
 luxx.State.prototype.bgLayer;
+
+/**
+ * @typedef {Object}
+ */
+luxx.GeocodeOptions;
```
Comment (tonio) [inline jsapi/externs/luxx.js]: those options don’t seems to be defined elsewhere

File: jsapi/src/main.js [hunk 3]
```diff
@@ -1094,3 +1105,48 @@ lux.WMSLayerFactory_ = function(config, opacity) {
...
+  var url = goog.Uri.parse(this.geocodeUrl);
+  Object.keys(obj).forEach(function(key) {
+    url.setParameterValue(key, obj[key]);
+  });
+  fetch(url.toString()).then(function(resp) {
+    return resp.json();
+  }).then(function(json) {
+    goog.asserts.assert(json.results.length, 'No address was found');
+    var result = json.results[0];
+    cb.call(null, ol.proj.transform([result.easting, result.northing],
```
Comment (tonio) [inline jsapi/src/main.js]: Is this transform really needed — or asked — ?

File: jsapi/examples/geocode.js (hunk 1 above)
Comment (pgiraud) [inline jsapi/examples/geocode.js]: Removed.

File: jsapi/src/main.js (hunk 3 above)
Comment (pgiraud) [inline jsapi/src/main.js]: Indeed, this adds unrequired complexity.

File: jsapi/externs/luxx.js (hunk 2 above)
Comment (pgiraud) [inline jsapi/externs/luxx.js]: Possible values added. This will help for future documentation.

**Pull request merged**